### PR TITLE
feat: pad protocol fee estimate for quotes that include RFQT orders

### PR DESCRIPTION
**Pull request by kimpers**
# Description

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [x] Add tests to cover changes as needed.
-   [x] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [x] Swap endpoints
    -   [x] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

File: src/services/swap_service.ts [hunk 1]
```diff
@@ -219,7 +221,39 @@ export class SwapService {
...
+            return isNativeOrder && hasTakerAddress;
+        });
+
+        // NOTE: Takers have a tendency to bump the gas price in order to speed up their trades
+        // for Native orders this often leads to an insufficient protocol fee being included and
+        // potential RFQT orders cannot be filled, so to avoid this we pad the protocol fee amount
+        // specifically when RFQT orders are included in the quote
+        if (hasRFQTOrders) {
+            const maxGasPricePadding = gasPrice.times(RFQT_PROTOCOL_FEE_GAS_PRICE_MAX_PADDING_MULTIPLIER);
+            const fastestGasPrice = await ethGasStationUtils.getGasPriceOrThrowAsync('fastest').catch(err => {
```
Comment (kimpers) [inline src/services/swap_service.ts:238]: This potential extra gas price lookup (if there's nothing in cache) adds some extra response time for RFQT orders but given that this is a short-term bandaid I think that is okay. Wdyt?

**Review by dekz — APPROVED** [review]

File: src/services/swap_service.ts (hunk 1 above)
Comment (dekz) [reply src/services/swap_service.ts:238]: It's ok since its an internal proxy cache and should be fast. And in case it fails it is handled well.

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy production

**Comment by kimpers** [comment]
deploy production

**Pull request merged**